usr/share/rear/layout/prepare/GNU/Linux/210_load_multipath.sh

```diff
@@ -34,15 +34,59 @@ blacklist {
     fi
 
     LogPrint "Activating multipath"
-    modprobe dm-multipath >&2
-    multipath >&2
-    if [ $? -ne 0 ] ; then
-        LogPrint "Failed to activate multipath, or no multipath device found."
-        rear_shell "Did you activate the multipath devices?"
-    else
-        LogPrint "multipath activated"
+    list_mpath_device=1
+    modprobe dm-multipath >&2 && LogPrint "multipath activated" || LogPrint "Failed to load dm-multipath module"
+
+    # Asking to the User what to do after multipath command return 1.
+    # It could be because no multipath device were found (sles11/rhel6)
+    # or a real problem in the multipath configuration.
+    prompt="Failed to get multipath device list or no multipath device found."
+
+    rear_workflow="rear $WORKFLOW"
+    unset choices
+    choices[0]="Multipath is not needed. Continue recovery."
+    choices[1]="Run multipath with debug options."
+    choices[2]="Enter into rear-shell to manually debug multipath."
+    choices[3]="Abort '$rear_workflow'"
+    choice=""
+    wilful_input=""
+
+    # looping on this menu while multipath failed to list device.
+    # Note: On sles11/rhel6, multipath failed if no multipath device is found.
+    while ! multipath ; do
+        echo
+        choice="$( UserInput -p "$prompt" -D "${choices[0]}" "${choices[@]}")"&& wilful_input="yes" || wilful_input="no"
+        case "$choice" in
+            (${choices[0]})
+                # continue recovery without multipath
+                is_true "$wilful_input" && LogPrint "User confirmed continuing without multipath" || LogPrint "Continuing '$rear_workflow' by default"
+                LogPrint "You should consider removing BOOT_ON_SAN parameter from your rear configuration file if you don't need multipath on this server."
+
+                # Avoid to list mpath device.
+                list_mpath_device=0
+                break
+                ;;
+            (${choices[1]})
+                # Run multipath in debug level
+                LogPrint "starting multipath -v3 (debug mode)"
+                multipath -v3
+                ;;
+            (${choices[2]})
+                # Exit to shell to debug multipathing issue
+                rear_shell "Do you want to go back to '$rear_workflow' ?"
+                ;;
+            (${choices[3]})
+                # Abort rear recovery
+                abort_recreate
+                Error "User chose to abort '$rear_workflow' in ${BASH_SOURCE[0]}"
+                ;;
+        esac
+    done
+
+    # Search and list mpath device.
+    if is_true $list_mpath_device ; then
+        LogPrint "Listing multipath device found"
         dmsetup ls --target multipath
```

Comment:
I think a little bit longer and I will not add one because:

The reason why
<pre>
LogPrint "$( cat qqq /etc/os-release QQQ )"
</pre>
shows only STDOUT to the user is that
the cat command does not output its STDERR on STDOUT
so that the cat SDTERR does not apperar in the arguments
for the LorPrint function.

With
<pre>
LogPrint "$( cat qqq /etc/os-release QQQ 2>&1 )"
</pre>
everything works perfectly - even in the right ordering:
<pre>
# usr/sbin/rear -v mkrescue
Relax-and-Recover 2.2 / Git
Using log file: /root/rear.master/var/log/rear/rear-e205.log
cat: qqq: No such file or directory
NAME="SLES"
VERSION="12-SP2"
VERSION_ID="12.2"
PRETTY_NAME="SUSE Linux Enterprise Server 12 SP2"
ID="sles"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:12:sp2"
cat: QQQ: No such file or directory
Creating disk layout
</pre>
and also in the log
<pre>
2017-08-24 13:46:31.781826067 Including init/default/030_update_recovery_system.sh
2017-08-24 13:46:31.783326447 cat: qqq: No such file or directory
NAME="SLES"
VERSION="12-SP2"
VERSION_ID="12.2"
PRETTY_NAME="SUSE Linux Enterprise Server 12 SP2"
ID="sles"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:12:sp2"
cat: QQQ: No such file or directory
2017-08-24 13:46:31.784369007 Including init/default/050_check_rear_recover_mode.sh
</pre>